Criollo/Source/CRConnection.m

```diff
@@ -217,7 +220,11 @@ - (void)didFinishResponseForRequest:(CRRequest *)request {
     CRConnection * __weak connection = self;
     [self.delegate connection:self didFinishRequest:request response:request.response];
     dispatch_async(self.isolationQueue, ^{
-        [connection.requests removeObject:request];
+        if (connection != nil) {
```

Comment:
No need for `nil` checks here. Obj-c allows for sending messages to `nil`. If `connection` is `nil` then the whole line is a no-op, Which makes for a nicer speed optimization than branching.

Reply:
The check-for-nil is to prevent calling `@syncronized (nil)`.

Reply:
👍